### PR TITLE
[MESH] set color ramp classification from metadata

### PR DESCRIPTION
**Pull request by vcloarec**

### CODE REVIEW

File: src/core/mesh/qgsmeshlayerutils.h [hunk 1]
```diff
@@ -282,6 +282,14 @@ class CORE_EXPORT QgsMeshLayerUtils
...
       const QVector<double> &verticalMagnitude,
       bool isRelative );
+
+    /**
+     * Changes scalar settings for classified scalar value
+     * \param meta the metadata where classification information is
+     * \param scalarSettings scalar settings that have to be modified
+     * \since QGIS 3.14
+     */
+    static void applyClassificationOnScalarSettings( const QgsMeshDatasetGroupMetadata &meta, QgsMeshRendererScalarSettings &scalarSettings );
```
Comment (PeterPetrik) [inline src/core/mesh/qgsmeshlayerutils.h]: Do we need to have it in public api? I suggest to move it as private method in qgsmeshlayer.cpp

File: src/app/mesh/qgsmeshrendereractivedatasetwidget.cpp [hunk 2]
```diff
@@ -185,6 +185,11 @@ QString QgsMeshRendererActiveDatasetWidget::metadata( QgsMeshDatasetIndex datase
   const auto options = gmeta.extraOptions();
   for ( auto it = options.constBegin(); it != options.constEnd(); ++it )
   {
+    if ( it.key() == QStringLiteral( "classification" ) )
+    {
+      msg += QStringLiteral( "<tr><td>Classified values</td></tr>" );
+      continue;
+    }
```
Comment (PeterPetrik) [inline src/app/mesh/qgsmeshrendereractivedatasetwidget.cpp]: i do not think it is necessary to treat it in special way..., i would remove this at all

File: src/core/mesh/qgsmeshlayerutils.cpp [hunk 3]
```diff
@@ -587,5 +587,93 @@ QVector<QVector3D> QgsMeshLayerUtils::calculateNormals( const QgsTriangularMesh
...
+        {
+          double scalarValue = ( boundClass.first() + boundClass.last() ) / 2;
+          item.color = colorRamp->color( ( scalarValue - minValue ) / diff );
+          if ( i != 0 && i < bounds.count() - 1 ) //The first and last labels are treated after
+          {
+            item.label = QString( ( "%1 - %2 %3" ) ).
+                         arg( QString::number( boundClass.first() ) ).
+                         arg( QString::number( boundClass.last() ) ).
+                         arg( units );
+          }
```
Comment (PeterPetrik) [inline src/core/mesh/qgsmeshlayerutils.cpp]: this should be taken from MDAL from `flag_meanings` attribute

File: src/core/mesh/qgsmeshlayerutils.h (hunk 1 above)
Comment (PeterPetrik) [inline src/core/mesh/qgsmeshlayerutils.h]: instead of const QgsMeshDatasetGroupMetadata &meta just pass const QString& with the classification?

**Comment by PeterPetrik** [comment]
before doing official PR, we need to merge MDAL PR and also add here the MDAL 0.5.92. otherwise the QGIS tests will fail...

File: src/app/mesh/qgsmeshrendereractivedatasetwidget.cpp (hunk 2 above)
Comment (vcloarec) [reply src/app/mesh/qgsmeshrendereractivedatasetwidget.cpp]: This metadata is very long and leads to a display issue on the styling widget (the width of the widget  take the width of the meta data line). Three solution :
- format the dataset to have it on several lines
- not display the classification at all
- only display that the dataset group is classified (so the user could know why the value are int and not the real value)

I have implemented the last one.

Comment (PeterPetrik) [reply src/app/mesh/qgsmeshrendereractivedatasetwidget.cpp]: ok

**Pull request closed without merging**